CRM/Contribute/BAO/Contribution.php

```diff
@@ -2366,7 +2367,7 @@ public function loadRelatedObjects($input, &$ids, $loadAll = FALSE) {
     // 5) hone up the individual functions to not use rely on this having been called
     // 6) deprecate like mad
     if ($loadAll) {
-      $ids = array_merge($this->getComponentDetails($this->id), $ids);
+      $ids = array_merge(self::getComponentDetails($this->id), $ids);
```

Comment:
@agileware-fj Francis is there a reason to swap from $this-> to self::?

Reply:
@seamuslee001 `getComponentDetails` is a static member function. I think the format `$this->` actually got the red squiggly line in the IDE because of that.

Is there some kind of coding standard in CiviCRM to the object scope operator for static member functions? Sounds confusing.

Reply:
@seamuslee001 This is correct. `getComponentDetails` is a static function and is called statically everywhere else. So `self::..` is correct.

Reply:
@agileware-fj - Looks like this has been refactored out now. The `if ($loadAll) {` part has vanished from the file.